GitCommands/Git/GitCommandHelpers.cs

```diff
@@ -776,94 +704,48 @@ public static string StartBisectCmd()
 
         private static string GetBisectCommand(GitBisectOption bisectOption)
         {
-            switch (bisectOption)
+            var args = new ArgumentBuilder
             {
-                case GitBisectOption.Good:
-                    return "bisect good";
-                case GitBisectOption.Bad:
-                    return "bisect bad";
-                case GitBisectOption.Skip:
-                    return "bisect skip";
-                default:
-                    throw new NotSupportedException(string.Format("Bisect option {0} is not supported", bisectOption));
-            }
+                "bisect",
+                bisectOption
+            };
+
+            return args.ToString();
         }
 
         public static string StopBisectCmd()
         {
             return "bisect reset";
         }
 
-        public static string RebaseCmd(string branch, bool interactive, bool preserveMerges, bool autosquash, bool autostash)
-        {
-            StringBuilder sb = new StringBuilder("rebase ");
-
-            if (interactive)
-            {
-                sb.Append(" -i ");
-                sb.Append(autosquash ? "--autosquash " : "--no-autosquash ");
-            }
-
-            if (preserveMerges)
-            {
-                sb.Append("--preserve-merges ");
-            }
-
-            if (autostash)
-            {
-                sb.Append("--autostash ");
-            }
-
-            sb.Append('"');
-            sb.Append(branch);
-            sb.Append('"');
-
-            return sb.ToString();
-        }
-
-        public static string RebaseRangeCmd(string from, string branch, string onto, bool interactive, bool preserveMerges, bool autosquash, bool autostash)
+        public static string RebaseCmd(string branch, bool interactive, bool preserveMerges, bool autosquash, bool autostash, string from = null, string onto = null)
         {
-            StringBuilder sb = new StringBuilder("rebase ");
-
-            if (interactive)
-            {
-                sb.Append(" -i ");
-                sb.Append(autosquash ? "--autosquash " : "--no-autosquash ");
-            }
-
-            if (preserveMerges)
-            {
-                sb.Append("--preserve-merges ");
-            }
-
-            if (autostash)
+            var args = new ArgumentBuilder
             {
-                sb.Append("--autostash ");
-            }
-
-            sb.Append('"')
-              .Append(from)
-              .Append("\" ");
-
-            sb.Append('"')
-              .Append(branch)
-              .Append("\"");
-
-            sb.Append(" --onto ")
-              .Append(onto);
+                "rebase",
+                { interactive, "-i" },
+                { interactive && autosquash, "--autosquash" },
+                { interactive && !autosquash, "--no-autosquash" },
+                { preserveMerges, "--preserve-merges" },
+                { autostash, "--autostash" },
+                from.QuoteNE(),
```

Comment:
there is no check if just one of from and onto is null. Should at least be a comment

Reply:
I've added a precondition:

```c#
if (from == null ^ onto == null)
{
    throw new ArgumentException(
        $"For arguments \"{nameof(from)}\" and \"{nameof(onto)}\", either both must have values, or neither may.");
}
```

Existing usages will be fine with this.